java/client/src/main/java/glide/api/commands/servermodules/GlideJson.java

```diff
@@ -0,0 +1,411 @@
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                        client, new String[] {JSON_SET, key, path, value, setCondition.getValkeyApi()})
+                .thenApply(r -> (String) r);
+    }
+
+    /**
+     * Sets the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path Represents the path within the JSON document where the value will be set. The key
+     *     will be modified only if <code>value</code> is added as the last child in the specified
+     *     <code>path</code>, or if the specified <code>path</code> acts as the parent of a new child
+     *     being added.
+     * @param value The value to set at the specific path, in JSON formatted GlideString.
+     * @param setCondition Set the value only if the given condition is met (within the key or path).
+     * @return A simple <code>"OK"</code> response if the value is successfully set. If value isn't
+     *     set because of <code>set_condition</code>, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, gs("doc"), , gs("."), gs("{'a': 1.0, 'b': 2}"), ConditionalChange.ONLY_IF_DOES_NOT_EXIST).get();
+     * assert value.equals("OK");
+     * }</pre>
+     */
+    public static CompletableFuture<String> set(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            @NonNull GlideString value,
+            @NonNull ConditionalChange setCondition) {
+        return executeCommand(
+                        client,
+                        new GlideString[] {gs(JSON_SET), key, path, value, gs(setCondition.getValkeyApi())})
+                .thenApply(r -> (String) r);
+    }
+
+    /**
+     * Retrieves the JSON value at the specified <code>path</code> stored at <code>key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param path The path within the JSON document.
+     * @return For JSONPath (path starts with <code>$</code>): Returns a stringifies JSON list of
+     *     bytes replies for every possible path, or a byte string representation of an empty array,
+     *     if path doesn't exist. For legacy path (path doesn't start with <code>$</code>): Returns a
+     *     byte string representation of the value in <code>path</code>. If <code>path</code> doesn't
+     *     exist, an error is raised. If <code>key</code> doesn't exist, returns null.
+     * @example
+     *     <pre>{@code
+     * String value = client.Json.set(client, "doc", "$").get();
+     * assert value.equals("{'a': 1.0, 'b': 2}");
+     * }</pre>
+     */
+    public static CompletableFuture<Object> get(BaseClient client, String key, String path) {
```

Comment:
```suggestion
    public static CompletableFuture<String> get(BaseClient client, String key, String path) {
```

Reply:
You will need to cast the result to String.

Reply:
TBD String vs GlideString